modules/top-level/plugins/byte-compile-lua-lib.nix

```
@@ -9,20 +9,32 @@ let
   builders = lib.nixvim.builders.withPkgs pkgs;
   inherit (import ./utils.nix lib) mapNormalizedPlugins;
 
+  # Applies a function to the derivation, but only if it's a lua module,
+  # otherwise returns it as is
+  mapLuaModule = f: drv: if pkgs.luaPackages.luaLib.hasLuaModule drv then f drv else drv;
+
   # Byte-compile only lua dependencies
   byteCompileDeps =
     drv:
-    drv.overrideAttrs (
-      prev:
-      lib.optionalAttrs (prev ? propagatedBuildInputs) {
-        propagatedBuildInputs = map byteCompile prev.propagatedBuildInputs;
-      }
-    );
+    lib.pipe drv [
+      (
+        drv:
+        drv.overrideAttrs (
+          prev:
+          # Process also dependencies
+          lib.optionalAttrs (prev ? dependencies) {
+            dependencies = map byteCompileDeps prev.dependencies;
+          }
+          // lib.optionalAttrs (prev ? propagatedBuildInputs) {
+            propagatedBuildInputs = map byteCompile prev.propagatedBuildInputs;
+          }
+        )
+      )
+      # 'toLuaModule' updates requiredLuaModules attr
+      # It works without it, but update it anyway for consistency
+      (mapLuaModule (drv: drv.lua.pkgs.toLuaModule drv))
```

Do we need `drv.lua`? If we had another way to reliably get the correct lua pkgs, we could do:

```suggestion
      (mapLuaModule lua.pkgs.toLuaModule)
```

No big deal if this is more correct.

---

Technically, various derivations can have different lua. But it won't work in runtime if the version is different (e.g. neovim is built with luajit, dependency is from lua5_2, import paths won't match). But I think we can take a lua from neovim-unwrapped as I pretty much always do. It can break only in very rare cases if the user for some reason overrode neovim package with another version of lua, but pkgs' neovim is standard luajit. I don't think we should consider this case, since doing this would be bad. On the other thought, nothing should break, because `toLuaModule` changes only passthru attributes, so let's take lua from pkgs' neovim-unwrapped.